packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1092,6 +1091,10 @@ export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
 	private readonly initialSnapshotDetailsP = new LazyPromise<ISnapshotDetails>(async () => {
 		// Sequence number of the snapshot.
 		let sequenceNumber: number | undefined;
+		// Check whether we need to fetch the snapshot first to load.
+		if (this.snapshotFetchRequired === undefined && this._baseSnapshot?.groupId !== undefined) {
+			this.snapshotFetchRequired = this.runtime.isSnapshotFetchRequired([this.id]);
```

Comment:
Is it better to cache this once, to run the check every time? I'm more curious as to which is better.

Reply:
In this logic, we are setting snapshotFetchRequired to be false, once we fetch it. So, it will not ask that question again to container runtime.

Reply:
That's a lazy promise, so we will run it once (at least once per data store)

Reply:
As per logic, we are only running it if the datastore has a gropupId